model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java

```diff
@@ -910,16 +910,8 @@ private Predicate[] getSearchOptionPredicateArray(String value, CriteriaBuilder
             orPredicates.add(builder.equal(builder.lower(from.get(FIRST_NAME)), value));
             orPredicates.add(builder.equal(builder.lower(from.get(LAST_NAME)), value));
         } else {
-            if (value.length() >= 2 && value.charAt(0) == '*' && value.charAt(value.length() - 1) == '*') {
-                // infix search
-                value = "%" + value.substring(1, value.length() - 1) + "%";
-            } else {
-                // default to prefix search
-                if (value.length() > 0 && value.charAt(value.length() - 1) == '*') {
-                    value = value.substring(0, value.length() - 1);
-                }
-                value += "%";
-            }
+            value = value.replace("*", "%");
+            if (value.isEmpty() || value.charAt(value.length() - 1) != '%') value += "%";
 
             orPredicates.add(builder.like(from.get(USERNAME), value));
             orPredicates.add(builder.like(from.get(EMAIL), value));
```

Comment:
I had a look and one can add a thirst parameter to `.like()` with specifying the escape character. I had a look around what other frameworks do, and they seem to have `\` as the default escape character. Still there seems to be some intrinsic behavior at least in MS SQL where one can use backets to escape parts of a `LIKE`, unless you explicitly state the escaping character - see https://www.mytecbits.com/microsoft/sql-server/escape-square-brackets-in-like.

So I'd suggest to use the third parameter to make the escaping explicit in the code, and safe from MS SQL intrinsic. As `\` is then used in multiple places, it IMHO calls for a constant to be defined in this class for this value.